cli/hooks/webpack.js

```diff
@@ -0,0 +1,163 @@
+// eslint-disable-next-line security/detect-child-process
+const { execSync } = require('child_process');
+const fs = require('fs-extra');
+const path = require('path');
+const semver = require('semver');
+const util = require('util');
+require('colors');
+
+const WebpackService = require('../lib/webpack/service');
+
+const MIN_APPCD_VERSION = '3.2.0';
+
+exports.id = 'ti.webpack';
+exports.init = (logger, config, cli) => {
+	let commandName;
+	let isWebpackEnabled = false;
+	let projectType;
+	let client;
+
+	cli.on('cli:command-loaded', (hookData) => {
+		const command = hookData.command;
+		commandName = command.name;
+	});
+
+	cli.on('cli:post-validate', () => {
+		projectType = getWebpackProjectType(cli.argv['project-dir']);
+		if (projectType === null) {
+			return;
+		}
+		isWebpackEnabled = true;
+		process.env.TI_USE_WEBPACK = true;
+
+		if (commandName !== 'build') {
+			return;
+		}
+
+		let appcdRootPath;
+		const isAppcCli = typeof process.env.APPC_ENV !== 'undefined';
+		if (isAppcCli) {
+			// we were invoked by appc-cli, load bundled daemon client.
+			appcdRootPath = resolveModuleRoot('appcd');
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find Appcelerator Daemon inside the appc-cli. Please make sure to use a recent appc-cli version. You can install/select it with  "appc use latest".');
+			}
+		} else {
+			// plain ti-cli, load daemon from global modules
+			const globalModulesPath = execSync('npm root -g').toString().replace(/\s+$/, '');
+			appcdRootPath = resolveModuleRoot('appcd', [ globalModulesPath ]);
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find global Appcelerator Daemon install. You can install it with "npm i appcd -g".');
+			}
+		}
+		ensureAppcdVersion(appcdRootPath, MIN_APPCD_VERSION, isAppcCli);
+		const AppcdClient = resolveAppcdClient(appcdRootPath);
+		client = new AppcdClient();
+		if (isAppcCli) {
+			client.appcd = path.resolve(appcdRootPath, '..', '.bin', 'appcd');
+		}
+	});
+
+	cli.on('build.pre.compile', {
+		priority: 800,
+		post(builder, callback) {
+			if (!isWebpackEnabled) {
+				return callback();
+			}
+
+			builder.useWebpack = true;
+
+			const webpackLogger = {
+				info(...args) {
+					logger.info(this.formatWithBadge(args));
+				},
+				debug(...args) {
+					logger.debug(this.formatWithBadge(args));
+				},
+				formatWithBadge(args) {
+					return `${' WEBPACK '.bgCyan.black} ${util.format(...args)}`;
+				}
+			};
+
+			const webpackService = new WebpackService({
+				client,
+				logger: webpackLogger,
+				cli,
+				builder,
+				projectType
+			});
+			Promise.resolve().then(async () => {
+				await webpackService.ensureDaemonIsRunning();
+				await webpackService.ensureWebpackIsRunning();
+				await webpackService.waitForCompilationComplete();
+				return callback();
+			}).catch(e => {
+				logger.error(e.stack);
+				callback(e);
```

Comment:
<!--
  0 failure: 
  1 warning:  cli/hooks/webpack...
  
  
  DangerID: danger-id-Danger;
  File: cli/hooks/webpack.js;
  Line: 96;
-->

- :warning: cli/hooks/webpack.js line 96 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)